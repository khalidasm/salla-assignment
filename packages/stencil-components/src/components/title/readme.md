# title-component



<!-- Auto Generated Below -->


## Properties

| Property | Attribute | Description | Type     | Default     |
| -------- | --------- | ----------- | -------- | ----------- |
| `text`   | `text`    |             | `string` | `undefined` |


## Dependencies

### Used by

 - [cart-view-component](../cart-view)
 - [confirmation-view-component](../confirmation-view)
 - [shipping-view-component](../shipping-view)

### Graph
```mermaid
graph TD;
  cart-view-component --> title-component
  confirmation-view-component --> title-component
  shipping-view-component --> title-component
  style title-component fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
