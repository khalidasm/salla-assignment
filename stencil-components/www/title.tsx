import { Component, h, Prop } from "@stencil/core";

@Component({
  tag: "title-component",
  shadow: true,
})
export class Title {
  @Prop() text: string;

  render() {
    const { text } = this;
    return (
      <h1 class="font-bold text-[18px] text-primary">
        {text}
      </h1>
    );
  }
}
