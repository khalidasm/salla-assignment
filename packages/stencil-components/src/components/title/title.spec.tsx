import { newSpecPage } from "@stencil/core/testing";
import { Title } from "./title";

describe("title-component", () => {
    it("renders with text prop", async () => {
        const page = await newSpecPage({
            components: [Title],
            html: '<title-component text="Hello World"></title-component>',
        });
        expect(page.root).toMatchSnapshot();
        const h1 = page.root.shadowRoot.querySelector("h1");
        expect(h1.className).toContain("font-bold");
        expect(h1.className).toContain("text-[18px]");
        expect(h1.className).toContain("text-primary");
        expect(h1.textContent).toBe("Hello World");
    });

    it("renders with empty text", async () => {
        const page = await newSpecPage({
            components: [Title],
            html: '<title-component text=""></title-component>',
        });
        expect(page.root).toMatchSnapshot();
        const h1 = page.root.shadowRoot.querySelector("h1");
        expect(h1.className).toContain("font-bold");
        expect(h1.className).toContain("text-[18px]");
        expect(h1.className).toContain("text-primary");
        expect(h1.textContent).toBe("");
    });
});
