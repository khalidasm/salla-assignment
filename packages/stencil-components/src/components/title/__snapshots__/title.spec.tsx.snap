// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`title-component renders with empty text 1`] = `
<title-component text="">
  <template shadowrootmode="open">
    <h1 class="font-bold text-[18px] text-primary"></h1>
  </template>
</title-component>
`;

exports[`title-component renders with text prop 1`] = `
<title-component text="Hello World">
  <template shadowrootmode="open">
    <h1 class="font-bold text-[18px] text-primary">
      Hello World
    </h1>
  </template>
</title-component>
`;
